### PR TITLE
Mobile view

### PR DESCRIPTION
**Pull request by binarydevteam**

### CODE REVIEW

**Comment by binarydevteam** [comment]
@brucebinary  I think all is done, just someone remove `sass/_ciq_doorbell.sass`, I don't know why?

**Comment by brucebinary** [comment]
@binarydevteam doorbell is a customer feedback service, so I separated it out from the library when prepping it to be deployed to npm

File: src/index.js [hunk 1]
```diff
@@ -42,9 +40,12 @@ class SmartChart extends React.Component {
 
     render() {
         const {children, isMobile} = this.props;
+
+        this.chart.setIsMobile(this.props.isMobile);
```
Comment (aminroosta) [inline src/index.js]: @shayanypn do not set state in render() method.
That's totally frowned up on in READ 😄

**Comment by brucebinary** [comment]
@binarydevteam remove it. It's no longer part of the library

File: src/store/ChartStore.js [hunk 2]
```diff
@@ -309,7 +309,7 @@ class ChartStore {
      * @param {bool} status if true, measn mobile mode is active
      */
     setIsMobile(status) {
-        this.isMobile = status;
+        this.isMobile = !status ? CIQ.isMobile : status;
```
Comment (brucebinary) [inline src/store/ChartStore.js]: I think it's better to have `CIQ.isMobile` set by the `charts.binary.com` app. I'll set this when I separate out the app. For the time being discard the changes from `index.js` and ChartStore.js`, and I'll have this merged.

File: src/index.js [hunk 3]
```diff
@@ -26,9 +26,11 @@ class SmartChart extends React.Component {
             activeSymbols = data.active_symbols;
         }
         this.chart.setActiveSymbols(activeSymbols);
+        this.mainStore.chart.onSymbolChange = this.props.onSymbolChange;
+    }
 
+    componentWillMount(){
```
[the remark this answers is not in the document]
Comment (binarydevteam) [reply src/index.js]: well `componentWillMount` run before `componentDidMount`. I need to initial amount of `isMobile` before the component render

[the remark this answers is not in the document]
Comment (brucebinary) [reply src/index.js]: Ah I didn't notice `Will` and `Did`.

Anyhow. With `isMobile` passed as a prop we don't need `setIsMobile` anymore.

File: src/index.js [hunk 4]
```diff
@@ -41,10 +38,11 @@ class SmartChart extends React.Component {
     }
 
     render() {
-        const {children, isMobile} = this.props;
+        const {children} = this.props;
+
         return (
             <MobxProvider store={this.mainStore}>
-                <Chart lang="en" isMobile={isMobile}>
```
Comment (brucebinary) [inline src/index.js]: use this instead

File: src/index.js [hunk 5]
```diff
@@ -41,10 +38,11 @@ class SmartChart extends React.Component {
     }
 
     render() {
-        const {children, isMobile} = this.props;
```
Comment (brucebinary) [inline src/index.js]: use this instead

File: src/store/ChartStore.js [hunk 6]
```diff
@@ -49,7 +49,7 @@ class ChartStore {
     @observable comparisonSymbols = [];
     @observable categorizedSymbols = [];
     @observable barrierJSX;
-    @observable isMobile;
+    isMobile;
```
Comment (brucebinary) [inline src/store/ChartStore.js]: remove this

**Pull request merged**